target/arm/cpu.c

```diff
@@ -153,13 +175,7 @@ static void cp_reg_reset(gpointer key, gpointer value, gpointer opaque)
     if (!ri->fieldoffset) {
         return;
     }
-
-#ifdef TARGET_CHERI
-    if (cpreg_field_is_cap(ri)) {
-        CPREG_FIELDCAP(&cpu->env, ri) = ri->capresetvalue;
-    } else
-#endif
-        if (cpreg_field_is_64bit(ri)) {
+    if (cpreg_field_is_64bit(ri)) {
```

Comment:
You might want to have an empty if before this for if has_special_capresetvalue is true? Otherwise if resetvalue doesn't match capresetvalue's cursor things will get weird.

Reply:
Good catch!